src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs

```diff
@@ -302,6 +315,22 @@ public override int GetHashCode()
                 return Hash.Combine(MetadataName, Hash.Combine(_containingModule, Hash.Combine(_namespaceName, arity)));
             }
 
+            internal override NamedTypeSymbol AsNativeInt(bool asNativeInt)
+            {
+                Debug.Assert(this.SpecialType == SpecialType.System_IntPtr || this.SpecialType == SpecialType.System_UIntPtr);
+
+                var other = asNativeInt == _isNativeInt ?
+                    this :
+                    new TopLevel(_containingModule, _namespaceName, name, arity, mangleName, isNativeInt: asNativeInt, _lazyErrorInfo, _lazyContainingNamespace, _lazyTypeId);
+
+                Debug.Assert(other.Equals(this));
```

Comment:
>(other.Equals(this) [](start = 28, length = 19)

It feels like there is something wrong with the fact that objects are equal, yet different to a degree that they cannot replace one another.

Reply:
That matches the behavior for symbols representing `dynamic` and `object`.

---
In reply to: [370286459](https://github.com/dotnet/roslyn/pull/40994#discussion_r370286459) [](ancestors = 370286459)